src/mpid/ch4/netmod/ofi/ofi_coll_select.h

```diff
@@ -94,4 +94,385 @@ MPL_STATIC_INLINE_PREFIX
     }
 }
 
+MPL_STATIC_INLINE_PREFIX
+    MPIDI_OFI_coll_algo_container_t * MPIDI_OFI_Gather_select(const void *sendbuf,
+                                                              int sendcount,
+                                                              MPI_Datatype sendtype,
+                                                              void *recvbuf,
+                                                              int recvcount,
+                                                              MPI_Datatype recvtype,
+                                                              int root,
+                                                              MPIR_Comm * comm,
+                                                              MPIR_Errflag_t * errflag,
+                                                              MPIDI_OFI_coll_algo_container_t *
+                                                              ch4_algo_parameters_container_in
+                                                              ATTRIBUTE((unused)))
+{
+    return (MPIDI_OFI_coll_algo_container_t *) & MPIDI_OFI_Gather_intra_binomial_cnt;
```

Comment:
Why are we casting the return value here? This is already the type we are dealing with.

Reply:
@raffenet ```MPIDI_OFI_Gather_intra_binomial_cnt``` is of type ```const MPIDI_OFI_coll_algo_container_t```, but function returns ```MPIDI_OFI_coll_algo_container_t *```. If we do not cast the return value here we get a warning. We could change the function prototypes, but it will be a large piece of differences to review. If the removing of the casting is critical we could introduce the appropriate infrastructure for the part 2 of collective selection.